win32ss/user/ntuser/nonclient.c

```diff
@@ -511,7 +511,44 @@ DefWndDoSizeMove(PWND pwnd, WORD wParam)
 	      RECT newRect = unmodRect;
 
 	      if (!iconic && !DragFullWindows) UserDrawMovingFrame( hdc, &sizingRect, thickframe );
-	      if (hittest == HTCAPTION) RECTL_vOffsetRect( &newRect, dx, dy );
+          if (hittest == HTCAPTION)
+          {
+              // Restore window size if it is snapped
+              if ((pwnd->InternalPos.NormalRect.left != 0 && pwnd->InternalPos.NormalRect.top != 0 &&
+                   pwnd->InternalPos.NormalRect.right != 0 && pwnd->InternalPos.NormalRect.bottom != 0) &&
+
+                  (pwnd->InternalPos.NormalRect.left != pwnd->rcWindow.left || pwnd->InternalPos.NormalRect.top != pwnd->rcWindow.top ||
+                   pwnd->InternalPos.NormalRect.right != pwnd->rcWindow.right || pwnd->InternalPos.NormalRect.bottom != pwnd->rcWindow.bottom))
+              {
+                  UserSetCursorPos((pwnd->InternalPos.NormalRect.left < 0 ? 0 : pwnd->InternalPos.NormalRect.left) + pt.x, pwnd->InternalPos.NormalRect.top + pt.y, 0, 0, FALSE);
+
+                  // Save normal size - it required when window unsnapped from one side and snapped to another holding mouse down
+                  origRect = pwnd->InternalPos.NormalRect;
+
+                  // Restore from maximized state
+                  if (Style & WS_MAXIMIZE)
+                  {
+                      co_IntSendMessage(UserHMGetHandle(pwnd), WM_SYSCOMMAND, SC_RESTORE, 0);
+                  }
+                  // Restore snapped to left/right place
+                  else
+                  {
+                      co_WinPosSetWindowPos(pwnd,
+                                            0,
+                                            pwnd->InternalPos.NormalRect.left,
+                                            pwnd->InternalPos.NormalRect.top,
+                                            pwnd->InternalPos.NormalRect.right - pwnd->InternalPos.NormalRect.left,
+                                            pwnd->InternalPos.NormalRect.bottom - pwnd->InternalPos.NormalRect.top,
+                                            0);
+                  }
+                  RECT empty = {0, 0, 0, 0};
+                  pwnd->InternalPos.NormalRect = empty;
```

Comment:
```suggestion
                  RECTL_vSetEmptyRect(&pwnd->InternalPos.NormalRect);
```

Reply:
Yes that is even better than what I proposed to make this C89 compatible!